src/envs/behavior_options.py

```diff
@@ -931,12 +1011,37 @@ def placeOntopObjectOptionPolicy(_state: State,
     return placeOntopObjectOptionPolicy
 
 
+def create_place_option_model(
+        plan: List[List[float]]) -> Callable[[State, "BehaviorEnv"], None]:
+    """Instantiates and returns a place option model function given an RRT
+    plan."""
+
+    def placeOntopObjectOptionModel(_init_state: State,
+                                    env: "BehaviorEnv") -> None:
+        rh_orig_grasp_postion = env.robots[0].parts["right_hand"].get_position(
+        )
+        rh_orig_grasp_orn = env.robots[0].parts["right_hand"].get_orientation()
+        target_pos = plan[-1][0:3]
+        target_orn = plan[-1][3:6]
+        env.robots[0].parts["right_hand"].set_position_orientation(
+            target_pos, p.getQuaternionFromEuler(target_orn))
+        env.robots[0].parts["right_hand"].force_release_obj()
+        # this is running a zero action to step simulator
+        env.step(np.zeros(17))
+        env.robots[0].parts["right_hand"].set_position_orientation(
+            rh_orig_grasp_postion, rh_orig_grasp_orn)
+
+    return placeOntopObjectOptionModel
+
+
 def place_ontop_obj_pos(
     env: "BehaviorEnv",
     obj: Union["URDFObject", "RoomFloor"],
     place_rel_pos: Array,
+    ret_option_model: bool = False,
     rng: Optional[Generator] = None,
-) -> Optional[Callable[[State, "BehaviorEnv"], Tuple[Array, bool]]]:
+) -> Optional[Union[Callable[[State, "BehaviorEnv"], Tuple[Array, bool]],
+                    Callable[[State, "BehaviorEnv"], None]]]:
```

Comment:
can you go through and make sure all the docstrings are correct now? for instance, the one for this method says "If the placement is infeasible, returns an indication to this effect (None). Otherwise, returns a function that can be stepped like an option to output actions at each timestep." but this is out-of-date, because now we have the possibility that it can return a model as well. can you describe in the comment exactly what's going on with this crazy-looking return type? (for all the controller fns, not just this one)

Reply:
+1